### PR TITLE
Deck functions, gather/shuffle/draw

### PR DESCRIPTION
**Pull request by ykeara**
(and slightly changed constructor to theoretically shuffle deck)

These might need to be changed in scope etc, but logic should be what we need.

### CODE REVIEW

File: src/deck.tsx [hunk 1]
```diff
@@ -4,9 +4,12 @@ import bind from 'bind-decorator';
 import * as React from 'react';
 import * as ReactDOM from 'react-dom';
 import { PlayingCard } from './card';
+import { CardValue} from './types';
 
 type DeckProps = {
 }
+var drawnCards: CardValue[] = new Array();
+var remainingCards: CardValue[] = new Array();
```
Comment (lionleaf) [inline src/deck.tsx:12]: We might want to scope these within the CardDeck? Assuming we might want multiple decks active at some point?

File: src/deck.tsx [hunk 2]
```diff
@@ -15,11 +18,37 @@ class CardDeck extends React.Component<{}, DeckProps>{
 
 	constructor( props: any )
 	{
+
         super( props );
-        for (var i = 0; i < 52; i++) {
+        for (let i = 0; i < 52; i++) {
+            remainingCards.push(i);
+            this.shuffle();
```
Comment (lionleaf) [inline src/deck.tsx]: Should the shuffle be called every iteration or is it enough to call it once at the end? Like you do in "gather"

**Review by lionleaf — APPROVED** [review]
Looks good. Some minor comments you can choose whether to address now or not :)

File: src/deck.tsx (hunk 2 above)
Comment (ykeara) [reply src/deck.tsx]: I'll push that fix, yea that's a mistake

File: src/deck.tsx (hunk 1 above)
Comment (ykeara) [reply src/deck.tsx:12]: I can go either way here, just depends what requirements are on for when we make cards. If they don't need access to the deck then yea we can lower the scope without issue

**Pull request merged**